ext/session/session.c

```diff
@@ -2659,6 +2659,11 @@ PHP_FUNCTION(session_start)
 
 	/* set options */
 	if (options) {
+		if (UNEXPECTED(HT_IS_PACKED(Z_ARRVAL_P(options)))) {
+			zend_argument_type_error(1, "must be of type array with keys as string");
```

Comment:
This seems more like a ValueError than a TypeError? As we are correctly passing an array.

Moreover, you are not checking this condition in the foreach loop (e.g. an array of the shape `["key" => "val1", "val2"]` would not error. As such the `HT_IS_PACKED(Z_ARRVAL_P(options))` pre-check seems a bit pointless? Just integrate it into the foreach loop (and change to FOREACH_STR_KEY_VAL as well).